### DISCUSSION TITLE
TreeWalkAdapter

### DISCUSSION THREAD

**Discussion by meiyasan** [doctrine/DoctrineBundle#1546] [Q&A] [answered]
Hello,

I am looking for some examples and advanced documentation about TreeWalker and SqlWalker, but I only find very limited information:
https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/cookbook/dql-custom-walkers.html

Would you perhaps have along the time some additional examples to share ? I get really hard time to understand this. Especially about TreeWalker and the way the recurrence occurs in the CountWalker exemple in case of parent variable ?

My goal would be to design a "eager" walker able to use parent feature to add join an select if not already set by user with a depth limit like 1 or 2 layers. This would be very useful in order to cache entities with nested association and translations.

**Comment by ostrolucky** [comment, marked as answer]
doctrine-bundle is not a best place to talk about this. Try https://github.com/doctrine/orm/discussions
